microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -63,42 +63,105 @@ def check_dpdk_build(self, case_name: str, node: Node) -> None:
         "python3-pyelftools",
         "python-pyelftools",
     ]
-
+    _redhat_packages = [
+        "gcc",
+        "make",
+        "git",
+        "tar",
+        "wget",
+        "dos2unix",
+        "psmisc",
+        "kernel-devel",
+        "numactl-devel.x86_64",
+        "librdmacm-devel",
+        "pkgconfig",
+        "libmnl-devel",
+        "elfutils-libelf-devel",
+        "python3-pip",
+    ]
+    _rte_target = "x86_64-native-linuxapp-gcc"
     _dpdk_github = "https://github.com/DPDK/dpdk.git"
+    _ninja_url = (
+        "https://github.com/ninja-build/ninja/releases/download/v1.10.2/ninja-linux.zip"
+    )
 
     def _install_dpdk_dependencies(self, node: Node) -> None:
         if isinstance(node.os, Ubuntu):
-            node.os.install_packages(self._ubuntu_packages)
+            for package in self._ubuntu_packages:
+                node.os.install_packages(package)
             self.log.info("Packages installed for Ubunutu")
+            self._execute_expect_zero(node, "pip3 install --upgrade meson")
+            self._execute_expect_zero(node, "mv /usr/bin/meson /usr/bin/meson.bak")
+            self._execute_expect_zero(node, "ln -s /usr/local/bin/meson /usr/bin/meson")
+            self._execute_expect_zero(node, "pip3 install --upgrade ninja")
+
         elif isinstance(node.os, Redhat) or isinstance(node.os, CentOs):
+            self._execute_expect_zero(
+                node, "yum update -y --disablerepo='*' --enablerepo='*microsoft*'"
+            )
             node.os.install_packages(
                 ["groupinstall", "'Infiniband Support'"], signed=False
             )  # todo gross hack to support groupinstall
+            for package in self._redhat_packages:
+                node.os.install_packages(package)
             result = node.execute(
                 "dracut --add-drivers 'mlx4_en mlx4_ib mlx5_ib' -f"
             )  # add mellanox drivers
             self.log.debug("\n".join([result.stdout, result.stderr]))
+            self._execute_expect_zero(node, "systemctl enable rdma")
+            self._execute_expect_zero(node, "pip3 install --upgrade meson")
+            self._execute_expect_zero(node, "ln -s /usr/local/bin/meson /usr/bin/meson")
+
+            self._execute_expect_zero(
+                node,
+                f"wget {self._ninja_url}",
+            )
+            self._execute_expect_zero(
+                node, "unzip ninja-linux.zip && mv ninja /usr/bin/ninja"
+            )
+            self._execute_expect_zero(node, "pip3 install --upgrade pyelftools")
+
+    def _execute_expect_zero_with_path(
```

Comment:
1. There is a method of result, `assert_exit_code`. It has similar function.
2. Make sure the stdout and stderr is not too long to output. The message will be saved with test results together.

Reply:
Added code to truncate the output in this wrapper.